File: eel_v/web/backend/table_chose.py
import eel
from os import getcwd

from web.backend.table_edit import Editor
from web.backend.local.dbOperations import DataBase
from web.backend.local.execute_command import Execute
from web.backend.local.ini_parser import config
from web.backend.table_create import TableCreate
from web.backend.search import Search

class TableChose:

    #инициализация
    def __init__(fles, db : DataBase):
        TableChose.db = db

    @eel.expose
    def process_string(text:str):
        return text.strip()[-5:-1] == ' Tru'

    @eel.expose
    def set_val(value):
        config.set('General', 'dark_theme', str(value))
        with open('./web/backend/user/settings.ini', 'w') as conf_file:
            config.write(conf_file)

    @eel.expose
    def search(target):
        Search(target, TableChose.db)

    @eel.expose
    def load_tables():
        eel.add_table(TableChose.db.make_command(f'select table_name from information_schema.tables where table_schema = "{TableChose.db.db}";', False))
    
    @eel.expose
    def delete_last_table():
        TableChose.make_command('drop table {0};'.format(TableChose.db.make_command(f'select table_name from information_schema.tables where table_schema = "{TableChose.db.db}";', False)[-1][0]))

    @eel.expose
    def open_table(name):
        Editor(TableChose.db, name)
    
    @eel.expose
    def make_command(command):
        return Execute(TableChose.db).handle(command)
    
    @eel.expose
    def create():
        TableCreate(TableChose.db)
    

File: eel_v/web/backend/javascript/table_chose.js
eel.load_tables()

eel.expose(add_table)
function add_table(names){
    let table = document.querySelector('table#main_table')

    for (let i = 0; i < names.length; i++){
        let tr = document.createElement('tr')
        let td = document.createElement('td')
        let btn = document.createElement('button')

        btn.textContent = names[i][0]
        btn.id = names[i][0]
        btn.className = 'btn'
        btn.onclick = on_table.bind(null, names[i][0])

        td.appendChild(btn)
        tr.appendChild(td)
        table.appendChild(tr)
    }
}

eel.expose(ask)
function ask(question){return confirm(question)}

eel.expose(alert_chose)
function alert_chose(text){alert(text)}

eel.expose(reload)
function reload(){clear();eel.load_tables()}

function delete_last(){
    eel.delete_last_table()
    reload()
}

function on_table(name){
    eel.open_table(name)
    window.location.href = "http://localhost:8000/frontend/table_edit.html"
}

function make_command(){
    var command = document.getElementById('command')
    eel.make_command(command.value)().then(function(value){
        if (value){
            window.location.href = 'http://localhost:8000/frontend/table_instance.html'
        }
    })

}

function clear(){
    let table = document.querySelector('table#main_table')

    table.remove()
    table = document.createElement('table')
    table.id = 'main_table'
    let div = document.querySelector('div')

    div.appendChild(table)
}

function search_value(){

    var search_target = document.getElementById('search_value')
    eel.search(search_target.value)
    window.location.href = 'http://localhost:8000/frontend/search_result.html'

}

function go_to_create(){
    eel.create()
    window.location.href = 'http://localhost:8000/frontend/table_create.html'
}

function switch_theme() {
    
    if (document.getElementById('stylesheet').href == 'http://localhost:8000/frontend/dark.css'){

        document.getElementById('stylesheet').href = 'table_chose.css'
        document.getElementById('colors').href = 'colors.css'
        eel.set_val(false)

    }
    else{

        document.getElementById('stylesheet').href = 'dark.css'
        document.getElementById('colors').href = ''
        eel.set_val(true)

    }

}

function onload(){

    var url = "../backend/user/settings.ini"
    var xhr = new XMLHttpRequest()

    xhr.open('GET', url)
    xhr.send()

    xhr.onreadystatechange = function(){
        if (xhr.readyState == 4) {
            if (xhr.status == 200) {
                eel.process_string(xhr.responseText)().then(function(value){
                    if (value){
                        switch_theme()
                    }
                })
            }
        }
    }

}
